compiler/rustc_resolve/src/lib.rs

```diff
@@ -223,11 +226,15 @@ enum ResolutionError<'a> {
     /// Error E0128: generic parameters with a default cannot use forward-declared identifiers.
     ForwardDeclaredGenericParam,
     /// ERROR E0770: the type of const parameters must not depend on other generic parameters.
-    ParamInTyOfConstParam(Symbol),
+    ParamInTyOfConstParam { name: Symbol, is_type: Option<ParamInTyOfConstParamIsType> },
     /// generic parameters must not be used inside const evaluations.
     ///
     /// This error is only emitted when using `min_const_generics`.
-    ParamInNonTrivialAnonConst { name: Symbol, is_type: bool },
+    ParamInNonTrivialAnonConst { name: Symbol, is_type: ParamInNonTrivialAnonConstIsType },
+    /// generic parameters must not be used inside enum discriminants.
+    ///
+    /// This error is emitted even with `generic_const_exprs`.
+    ParamInEnumDiscriminant { name: Symbol, is_type: ParamInEnumDiscriminantIsType },
```

Comment:
Should NonTrivialConstArg take a NoConstantGenericsReason instead?

Reply:
If we removed this variant and added the reason to `ParamInNonTrivialConstArg` (and I guess renamed it to `ParamInConcreteAnonConst`) I'd also have to introduce another `ParamKind` enum separate from the ones in the `errors.rs` file I think. I don't feel like that's worth it but if you feel strongly I can do that.